src/storage/ducklake_field_data.cpp

```diff
@@ -207,6 +209,13 @@ unique_ptr<DuckLakeFieldId> DuckLakeFieldId::RemoveField(const vector<string> &c
 	for (idx_t child_idx = 0; child_idx < children.size(); child_idx++) {
 		auto &child = *children[child_idx];
 		if (StringUtil::CIEquals(child.Name(), column_path[depth])) {
+			if (depth == 1 && column_path.size() == 2 &&
+			    ((type.id() == LogicalTypeId::MAP &&
```

Comment:
We only need to check the type id here - we can never drop anything from a map or a list